boards/stm32f412gdiscovery/src/main.rs

```diff
@@ -179,6 +180,7 @@ unsafe fn set_pin_primary_functions(
     syscfg: &stm32f412g::syscfg::Syscfg,
     i2c1: &stm32f412g::i2c::I2C,
     gpio_ports: &'static stm32f412g::gpio::GpioPorts<'static>,
+    system_frequency: usize,
```

Comment:
```suggestion
    peripheral_clock_frequency: usize,
```

Maybe? `system` sounds very generic, and APB to me makes it seem like this is the general peripheral bus clock.